helix-core/src/main/java/org/apache/helix/task/TaskDriver.java

```diff
@@ -526,6 +524,123 @@ public void enqueueJobs(final String queue, final List<String> jobs,
     }
   }
 
+  /**
+   * Add task to a running (IN-PROGRESS) job with default timeout
```

Comment:
Not necessarily a running job, people can add task to NOT_STARTED jobs too, right?

Reply:
Add a task to a job, if a job is already completed, this operation has no effect?

Reply:
Added not started also. Yes if it is completed, then the add operation will be failed.